pages/content/amp-dev/documentation/guides-and-tutorials/optimize-measure/amp_to_pwa.md

```diff
@@ -0,0 +1,204 @@
+
+
+# Create a Web App Manifest
+
+The [web app manifest ](https://developers.google.com/web/fundamentals/web-app-manifest/)is a simple JSON file that tells the browser about your web application 
+and how it should behave when 'installed' on the user's mobile device or desktop. Having a manifest is required by many browsers to show the 
+[Add to Home Screen prompt](https://developers.google.com/web/fundamentals/app-install-banners/).
+
+Add a file titled `manifest.json` to your repository with the following code:
+
+
+[sourcecode:JSON]
+{
+ "short_name": "LyLy",
+ "name": "Lyrical Lyghtning",
+ "icons": [
+   {
+     "src": "./images/amplogo192.png",
+     "type": "image/png",
+     "sizes": "192x192"
+   },
+   {
+     "src": "./images/amplogo512.png",
+     "type": "image/png",
+     "sizes": "512x512"
+   }
+ ],
+ "start_url": "/index.html",
+ "background_color": "#222325",
+ "display": "standalone",
+ "scope": "/",
+ "theme_color": "#222325"
+}
+[/sourcecode]
+
+
+# Add the AMP Service Worker
+
+A service worker is a script that your browser runs in the background, 
+separate from a web page, that extends the browsers features by caching requests to improve performance and providing offline functionality. 
+Building a service worker from scratch is possible but time consuming. 
+Libraries like Workbox help, but AMP goes one step further by offering the [AMP Service Worke](https://github.com/ampproject/amp-sw)r, 
+in which AMP automates a lot of steps directly, including the caching of AMP Scripts, assets and documents.
+
+The AMP Service Worker automatically [caches AMP scripts](https://github.com/ampproject/amp-sw/tree/master/src/modules/amp-caching) 
+and [documents](https://github.com/ampproject/amp-sw/tree/master/src/modules/document-caching) as user requests them, after installing it. 
+We'll start by adding the basic AMP Service Worker.
+
+
+## Create the service worker file
+
+Create a file called `sw.js` and add the following code:
+
+[sourcecode:js]
+importScripts('https://cdn.ampproject.org/sw/amp-sw.js');
+AMP_SW.init();
+[/sourcecode]
+
+With just two lines of code, this imports the AMP Service Worker into your Service Worker and initializes it.
+
+## Auto-install your service worker on your AMP pages
+
+AMP websites use the [`<amp-install-serviceworker>`](https://www.ampproject.org/docs/reference/components/amp-install-serviceworker) 
+component to install the service worker in the browser's background, while the user is enjoying your content.
+
+Place the required script tag in the head of `index.html` and the `<amp-install-serviceworker>` element inside the `<body>`:
+
+[sourcecode:html]
+…
+<script async custom-element="amp-install-serviceworker" src="https://cdn.ampproject.org/v0/amp-install-serviceworker-0.1.js"></script>
+…
+</head>
+<body>
+   <amp-install-serviceworker src="/sw.js"
+           data-iframe-src="install-sw.html"
+           layout="nodisplay">
+   </amp-install-serviceworker>
+...
+[/sourcecode]
+
```

Comment:
Suggestion: add a callout explaining that the service worker should be served from the root

```
[tip type="important"]
 **Important –** the service worker needs to be served from the root directory (`/serviceworker.js`) to be able to cache all the content of your site.
 [/tip]
```